### ISSUE TITLE
optimization visualization

### ISSUE THREAD

**Issue by nnop** [nnop/notes#7]
- [Wiki - Test functions for optimization](https://en.wikipedia.org/wiki/Test_functions_for_optimization)
- [Scipy - 2.7. Mathematical optimization: finding minima of functions](http://www.scipy-lectures.org/advanced/mathematical_optimization/)
- [Y2B - Visualize Back Propagation](https://www.youtube.com/watch?v=7HZk7kGk5bU)
- [Visualizing and Animating Optimization Algorithms with Matplotlib](http://louistiao.me/posts/notebooks/visualizing-and-animating-optimization-algorithms-with-matplotlib/)
- [The Concept of Conjugate Gradient Descent in Python](http://ikuz.eu/2015/04/15/the-concept-of-conjugate-gradient-descent-in-python/)
- [Visualize Backpropagation Algorithms](http://neupy.com/versions/0.1.4/2015/07/04/visualize_backpropagation_algorithms.html)
